polkadot/xcm/src/v3/multiasset.rs

```diff
@@ -517,9 +516,16 @@ impl MaxEncodedLen for MultiAssets {
 }
 
 impl Decode for MultiAssets {
-	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, parity_scale_codec::Error> {
-		Self::from_sorted_and_deduplicated(Vec::<MultiAsset>::decode(input)?)
-			.map_err(|()| "Out of order".into())
+	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
+		let number_of_assets: u32 = <codec::Compact<u32>>::decode(input)?.into();
+		if number_of_assets > MAX_ITEMS_IN_MULTIASSETS as u32 {
+			return Err(codec::Error::from("Max MultiAssets exceeded"))
+		}
+		Self::from_sorted_and_deduplicated(codec::decode_vec_with_len(
+			input,
+			number_of_assets as usize,
+		)?)
+		.map_err(|()| "Out of order".into())
```

Comment:
You can use the `BoundedVec::<..>::decode(..)?.into_inner()` here again, or? Anyway it should both work.